cyclonedx/model/__init__.py

```diff
@@ -767,6 +769,28 @@ def deserialize(cls, o: Any) -> 'XsUri':
                 f'XsUri string supplied does not parse: {o!r}'
             ) from err
 
+    @classmethod
+    def make_bom_link(
+        cls,
+        serial_number: Union[UUID, str],
+        version: int = 1,
+        bom_ref: Optional[Union[str, BomRef]] = None
+    ) -> 'XsUri':
+        """
+        Generate a BOM-Link URI.
+
+        Args:
+            serial_number: The unique serial number of the BOM.
+            version: The version of the BOM. The default version is 1.
+            bom_ref: The unique identifier of the component, service, or vulnerability within the BOM.
+
+        Returns:
+            XsUri: Instance of XsUri with the generated BOM-Link URI.
+        """
+        bom_ref_part = f'#{str(bom_ref).replace("%", "%25").replace("#", "%23")}' if bom_ref else ''
```

Comment:
as [you suggested](https://github.com/CycloneDX/cyclonedx-python-lib/pull/728#discussion_r1819326485),   
better utilize `urllib.parse.quote` here. this should prevent a lot of issues in the future.